gateway-service/src/main/java/org/zowe/apiml/gateway/cache/LoadBalancerCache.java

```diff
@@ -51,20 +54,50 @@ public LoadBalancerCache(CachingServiceClient cachingServiceClient) {
      */
     public boolean store(String user, String service, LoadBalancerCacheRecord loadBalancerCacheRecord) {
         if (remoteCache != null) {
-            try {
-                remoteCache.create(new CachingServiceClient.KeyValue(getKey(user, service), mapper.writeValueAsString(loadBalancerCacheRecord)));
-                log.debug("Stored record to remote cache for user: {}, service: {}, record: {}", user, service, loadBalancerCacheRecord);
-            } catch (CachingServiceClient.CachingServiceClientException e) {
-                log.debug("Failed to store record for user: {}, service: {}, record {}, with exception: {}", user, service, loadBalancerCacheRecord, e);
-            } catch (JsonProcessingException e) {
-                log.debug("Failed to serialize record for user: {}, service: {}, record {},  with exception: {}", user, service, loadBalancerCacheRecord, e);
-            }
+            storeToRemoteCache(user, service, loadBalancerCacheRecord);
         }
         localCache.put(getKey(user, service), loadBalancerCacheRecord);
         log.debug("Stored record to local cache for user: {}, service: {}, record: {}", user, service, loadBalancerCacheRecord);
         return true;
     }
 
+    private void storeToRemoteCache(String user, String service, LoadBalancerCacheRecord loadBalancerCacheRecord) {
+        try {
+        String serializedRecord = mapper.writeValueAsString(loadBalancerCacheRecord);
+        CachingServiceClient.KeyValue toStore = new CachingServiceClient.KeyValue(getKey(user, service), serializedRecord);
+            createToRemoteCache(user, service, loadBalancerCacheRecord, toStore);
+        } catch (JsonProcessingException e) {
+            log.debug("Failed to serialize record for user: {}, service: {}, record {},  with exception: {}", user, service, loadBalancerCacheRecord, e);
+        }
+    }
+
+    private void createToRemoteCache(String user, String service, LoadBalancerCacheRecord loadBalancerCacheRecord, CachingServiceClient.KeyValue toStore) {
+        try {
+        remoteCache.create(toStore);
+        log.debug("Created record to remote cache for user: {}, service: {}, record: {}", user, service, loadBalancerCacheRecord);
+        } catch (CachingServiceClientException createException) {
+            if (isCausedByCacheConflict(createException)) {
+                updateToRemoteCache(user, service, loadBalancerCacheRecord, toStore);
```

Comment:
Is the overwrite of the currently selected, the best solution?

Reply:
I feel it is at the moment. We don't have optimizations, but they are tricky in this case.